lib/models/collection.ts

```diff
@@ -0,0 +1,47 @@
+import { Error, ErrorJSON } from './error';
+import { Item, ItemJSON } from './item';
+import { Link, LinkJSON } from './link';
+import { Links } from './links';
+import { Queries } from './queries';
+import { Query, QueryJSON } from './query';
+import { Template, TemplateJSON } from './template';
+
+export interface CollectionJSON {
+    version: string;
+    href: string;
+    links: Link[];
+    items: Item[];
+    queries: Query[];
+    template: Template;
+    error: Error;
+}
+
+export interface CollectionAPI {
+    parse(collection: CollectionJSON): void;
+    link(rel: string): Link;
+}
+
+export interface Collection extends CollectionJSON, CollectionAPI {}
+
+export abstract class CollectionBase implements CollectionAPI {
```

Comment:
This should implement Collection interface with the (CollectionData and CollectionAPI)